pep-0701.rst

```diff
@@ -302,33 +304,85 @@ will be tokenized as::
     RBRACE - '}'
     FSTRING_MIDDLE - ' more words '
     LBRACE - '{'
     NAME - 'c'
     PLUS - '+'
     NAME - 'd'
+    OP - '='
     RBRACE - '}'
-    FSTRING_END - ' final words' (without the end quote)
+    FSTRING_MIDDLE - ' final words'
+    FSTRING_END - "'"
 
 while ``f"""some words"""`` will be tokenized simply as::
 
     FSTRING_START - 'f"""'
-    FSTRING_END - 'some words'
-
-One way existing lexers can be adapted to emit these tokens is to incorporate a stack of "lexer modes"
-or to use a stack of different lexers. This is because the lexer needs to switch from "regular Python
-lexing" to "f-string lexing" when it encounters an f-string start token and as f-strings can be nested,
-the context needs to be preserved until the f-string closes. Also, the "lexer mode" inside an f-string
-expression part needs to behave as a "super-set" of the regular Python lexer (as it needs to be able to 
-switch back to f-string lexing when it encounters the ``}`` terminator for the expression part as well
-as handling f-string formatting and debug expressions). Of course, as mentioned before, is not possible to
-provide a precise specification of how this should be done as it will depend on the specific implementation
-and nature of the lexer to be changed.
-
-The specifics of how (or if) the ``tokenize`` module will emit these tokens (or others) and what
-is included in the emitted tokens are left out of this document and must be decided later in a regular
-CPython issue.
+    FSTRING_MIDDLE - 'some words'
+    FSTRING_END - '"""'
+  
+Changes to the tokenize module
+------------------------------
+
+The ``tokenize`` module will be adapted to emit these tokens as described in the previous section
+when parsing f-strings so tools can take advantage of this new tokenization schema and avoid having
+to implement their own f-string tokenizer and parser.
+
+How to produce these new tokens
+-------------------------------
+
+One way existing lexers can be adapted to emit these tokens is to incorporate a
+stack of "lexer modes" or to use a stack of different lexers. This is because
+the lexer needs to switch from "regular Python lexing" to "f-string lexing" when
+it encounters an f-string start token and as f-strings can be nested, the
+context needs to be preserved until the f-string closes. Also, the "lexer mode"
+inside an f-string expression part needs to behave as a "super-set" of the
+regular Python lexer (as it needs to be able to switch back to f-string lexing
+when it encounters the ``}`` terminator for the expression part as well as
+handling f-string formatting and debug expressions). For reference, here is a
+draft of the algorithm to modify a CPython-like tokenizer to emit these new
+tokens:
+
+1. If the lexer detects that an f-string is starting (by detecting the letter
+   'f/F' and one of the possible quotes) keep advancing until a character that is
+   not equal to the first character after the 'f/F' and emit a ``FSTRING_START`` token
```

Comment:
I don't think this is right, since `f""""a"""` evaluates to `"a` (so the last `"` shouldn't be part of the FSTRING_START token).

Reply:
Oh, good point. Will change this.

Thanks for the catch!

Reply:
Also, there are raw f-strings, so the next character after the `f` could be an `r`. I'd be OK with omitting that since this is an informal description of the algorithm though.

Reply:
Yeah, this is far for complete because otherwise would be too long. There are many details missing like how to capture debug expressions and what data needs to be captured on push/pop from the stack.